app/auth/vault_auth.rb

```diff
@@ -1,74 +1,49 @@
 class VaultAuth < Auth
+  def self.vault_url
+    ENV['VAULT_URL']
+  end
+
   def self.enabled?
-    ENV['VAULT_URL'].present?
+    vault_url.present?
   end
 
   def login
     authenticate
   end
 
   def authenticate
-    res = lookup
-    username = res.dig("data", "meta", "username")
-    @current_user = User.find_or_create_by!(name: username) if username
+    user = User.new(name: "vault-user-#{vault_token.hash.to_s(16)}")
```

Comment:
So when vault auth is used, `User` is not stored in DB, correct? I think that is fine because all authentication and authorization are done by Vault basically and Barcelona does not need to store users.
One concern here is, when we want to audit (who did what), there's no indication to identify "who". So I suggest to include `data.meta.username` field in the user name. 
What do you think?

Reply:
That sounds like a very good idea.

Reply:
@k2nr I revisited this, but it turns out the `data.meta.username` is in the reply from authenticating with vault (i.e. the reply to sending the github token to vault). Since we are doing this from the CLI now, we do not send this data to vault (we only provide barcelona with the client token).

I tried looking for it but vault does not seem to have an API to interrogate the user name from a client_token. This seems to make sense because there may be some auth methods where there are no user names (token for example)